al_servicec.js

```diff
@@ -65,22 +66,17 @@ class AimsC extends m_alUtil.RestServiceClient {
     }
 
     _isTokenExpired(aimsToken) {
-        return aimsToken.authentication.token_expiration > (Date.now()/1000 + 600);
+        return aimsToken.authentication.token_expiration <= (Date.now()/1000 + 600);
     }
 
     _isTokenMemCached() {
-        return this._aimsResponse && this._isTokenExpired(this._aimsResponse);
+        return (this._aimsResponse != undefined) && !this._isTokenExpired(this._aimsResponse);
     }
 
     _isTokenFileCached() {
         var filename = this._tokenCacheFile;
-        fs.readFile(filename, (readError, fileContent) => {
-            if (readError) {
-                if ( readError.code !== 'ENOENT' ) {
-                    fs.unlinkSync(filename);
-                }
-                return false;
-            }
+        try {
```

Comment:
Please double-check if we could have one `try..catch` here and reason about exceptions type: read or JSON parse.